config/dws/nnf-ruleset.yaml

```diff
@@ -46,7 +46,7 @@ spec:
       - key: "^capacity$"
         type: "string"
         pattern: "^\\d+(KiB|KB|MiB|MB|GiB|GB|TiB|TB)$"
-        isRequired: true
+        isRequired: false
```

Comment:
How does this affect when you ask for xfs with a capacity of 0 or something similar?

Reply:
That's handled in the persistent storage reconciler. It makes sure that every directive has capacity specified except when it's lustre and StandaloneMgtPoolName is specified. I guess my error message for the second case is wrong though. It shouldn't mention "lustre".

```
	if argsMap["type"] == "lustre" && len(pinnedProfile.Data.LustreStorage.StandaloneMGTPoolName) > 0 {
		if _, exists := argsMap["capacity"]; exists {
			return ctrl.Result{}, dwsv1alpha2.NewResourceError("").WithUserMessage("creating persistent MGT does not accept 'capacity' argument").WithFatal().WithUser()
		}
		...
	} else {
		if _, exists := argsMap["capacity"]; !exists {
			return ctrl.Result{}, dwsv1alpha2.NewResourceError("").WithUserMessage("creating persistent Lustre requires 'capacity' argument").WithFatal().WithUser()
		}
	}
```

Your original question of 0 capacity made me want to try that. We don't handle that case well for jobdw or create_persistent. It just hangs in proposal state. Maybe I'll add on a fix for that bug to this PR since it's in the area.